### PR TITLE
Cmr 7979 cache echo token in browse scalar 

### PR DESCRIPTION
**Pull request by isja17**
instead of retrieving the echo token from parameter store every time, we now cache it.

### CODE REVIEW

**Comment by brianalexander** [comment]
Here are my thoughts:

1. Why use Redis?  
We don't need to cache the token in Redis.  Doing the cacheing in memory will be more performant, save resources, and be more secure.
2. What happens if the token changes?  
Right now, the token is being cached in Redis and is only updated when the token expires (approx 24 hours).  What happens if the token changes? Is it possible for them to be out of sync?

A good flow would be:
1. Get the token when the application starts
2. If a request fails because the token is bad (Authorization 401 error?), request a new token, update the in-memory token, re-run the failed request.

**Review by brianalexander — COMMENTED** [review]
Here are my thoughts:

1. Why use Redis?  
We don't need to cache the token in Redis.  Doing the cacheing in memory will be more performant, save resources, and be more secure.
2. What happens if the token changes?  
Right now, the token is being cached in Redis and is only updated when the token expires (approx 24 hours).  What happens if the token changes? Is it possible for them to be out of sync?

A good flow would be:
1. Get the token when the application starts
2. If a request fails because the token is bad (Authorization 401 error?), request a new token, update the in-memory token, re-run the failed request.

**Review by brianalexander — APPROVED** [review]

**Review by jaybarra — APPROVED** [review]

File: browse-scaler/src/cmr.js [hunk 1]
```diff
@@ -1,27 +1,35 @@
 const fetch = require('node-fetch');
 const { getSecureParam } = require('./util');
-
 const config = require ('./config');
+const { setValue, getValue } = require ('./in-memory-cache')
+var authorizationToken;
```
Comment (ygliuvt) [inline browse-scaler/src/cmr.js:5]: remove this line.

File: browse-scaler/src/in-memory-cache.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    function getInstance() {
+        return _data;
+    }
+
+    return {
+      getInstance: getInstance
+    };
+  }());
+
+exports.setValue = (key, item) => { InMemoryCache.getInstance() [key] = item;}
```
Comment (ygliuvt) [inline browse-scaler/src/in-memory-cache.js:13]: change `item` to `value`.

File: browse-scaler/src/in-memory-cache.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+exports.getValue = (id) => { return InMemoryCache.getInstance() [id]}
```
Comment (ygliuvt) [inline browse-scaler/src/in-memory-cache.js:15]: change `id` to `key`.

**Review by ygliuvt — APPROVED** [review]
approved with minor change comments.

**Pull request merged**